pkg/server/list_users_test.go

```diff
@@ -400,6 +400,62 @@ func TestListUsers_Deadline(t *testing.T) {
 		require.Len(t, resp.GetUsers(), 1)
 	})
 
+	t.Run("return_no_error_and_partial_results_if_throttled_until_deadline", func(t *testing.T) {
+		ds := memory.New()
+		t.Cleanup(ds.Close)
+
+		modelStr := `
+			model
+				schema 1.1
+			type user
+
+			type group
+			relations
+				define member: [user, group#member]
+
+			type document
+			relations
+				define viewer: [user, group#member]`
+
+		tuples := []string{
+			"document:1#viewer@user:jon", // Observed before first dispatch
+			"document:1#viewer@group:eng#member",
+			"group:eng#member@group:backend#member",
+			"group:backend#member@user:tyler", // Requires two dispatches, gets throtled
+		}
+
+		storeID, model := test.BootstrapFGAStore(t, ds, modelStr, tuples)
+		t.Cleanup(ds.Close)
+
+		deadline := 30 * time.Millisecond
+
+		s := MustNewServerWithOpts(
+			WithDatastore(ds),
+			WithListUsersDeadline(deadline),
+			WithListUsersDispatchThrottlingEnabled(true),
+			WithListUsersDispatchThrottlingThreshold(1),          // Applies throttling after first dispatch
+			WithListUsersDispatchThrottlingFrequency(2*deadline), // Forces time-out when throttling occurs
+		)
+		t.Cleanup(s.Close)
+
+		resp, err := s.ListUsers(ctx, &openfgav1.ListUsersRequest{
+			StoreId:              storeID,
+			AuthorizationModelId: model.GetId(),
+			Object: &openfgav1.Object{
+				Type: "document",
+				Id:   "1",
+			},
+			Relation: "viewer",
+			UserFilters: []*openfgav1.UserTypeFilter{
+				{Type: "user"},
+			},
+		})
+
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+		require.Len(t, resp.GetUsers(), 1)
+	})
```

Comment:
Just to confirm my understanding - this test is asserting on existing behavior, not on components that you added in this PR

Reply:
Yes, correct. It was an attempt to improve code coverage for this PR but is also a test that should have existed in the first place. I can remove in this PR if you'd prefer.

Reply:
No, it's ok